plugins/aea-cli-ipfs/aea_cli_ipfs/ipfs_utils.py

```diff
@@ -303,26 +305,42 @@ def remove_unpinned_files(self) -> None:
                 f"Error while performing garbage collection: {str(e)}"
             ) from e
 
-    def download(self, hash_id: str, target_dir: str, fix_path: bool = True) -> str:
+    def download(
+        self,
+        hash_id: str,
+        target_dir: str,
+        fix_path: bool = True,
+        attempts: int = 5,
+    ) -> str:
         """
-        Download dir by it's hash.
+        Download dir by its hash.
 
         :param hash_id: str. hash of file to download
         :param target_dir: str. directory to place downloaded
         :param fix_path: bool. default True. on download don't wrap result in to hash_id directory.
+        :param attempts: int. default 5. How often to attempt the download.
         :return: downloaded path
         """
+
         if not os.path.exists(target_dir):  # pragma: nocover
             os.makedirs(target_dir, exist_ok=True)
 
         if os.path.exists(os.path.join(target_dir, hash_id)):  # pragma: nocover
             raise DownloadError(f"{hash_id} was already downloaded to {target_dir}")
 
-        try:
-            self.client.get(hash_id, target_dir)
-        except ipfshttpclient.exceptions.StatusError as e:
-            raise DownloadError(f"error on download {str(e)}") from e
         downloaded_path = str(Path(target_dir) / hash_id)
+        while attempts:
+            attempts -= 1
+            try:  # download to tmp_dir in case of midway download failure
+                with tempfile.TemporaryDirectory() as tmp_dir:
+                    self.client.get(hash_id, tmp_dir)
+                    download_path = Path(tmp_dir) / hash_id
+                    shutil.copytree(download_path, downloaded_path)
```

Comment:
probably name as target dir?

Reply:
I did that in a subsequent PR. Those I will get back to later this week as I need to revisit the test PR first.